src/Raven.Server/Documents/Queries/Parser/QueryScanner.cs

```diff
@@ -531,6 +531,78 @@ public bool FunctionBody()
 
         }
 
+        public bool TimeSeriesFunctionBody()
```

Comment:
This is duplicated, no?
Merge this with the JS body but accept the `{` vs `(` as param?

Reply:
If I'll accept open parentheses and closing parentheses as parameters, then I won't be able to use them as switch cases anymore (non constant values).

I can handle that inside the default case, i.e :
```
default:
	if (_q[_pos] == open)
	{
		nested++;
		break;
	}
	if (_q[_pos] == close && --nested == 0)
	{
		_pos += 1;
		_tokenStart = original;
		_tokenLength = _pos - original;
		return true;
	}
	break;
```
but that will affect performance, wouldn't it ?

Reply:
Then add a `isCurly` and do case for `{` and `(` and jump out if not? 
Honestly, though, I doubt that it would matter.

Reply:
done